test/test_out_dtype_op.py

```diff
@@ -0,0 +1,121 @@
+# Owner(s): ["module: functorch"]
+import torch
+import torch._dynamo
+import torch._export
+from torch._higher_order_ops.out_dtype import out_dtype
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.testing import FileCheck
+
+
+class TestOutDtypeOp(TestCase):
+    def test_out_dtype_make_fx(self):
+        class M(torch.nn.Module):
+            def __init__(self, weight):
+                super().__init__()
+                self.weight = weight
+
+            def forward(self, x):
+                return out_dtype(
+                    torch.ops.aten.mm.default, torch.int32, x, self.weight
+                )
+
+        weight = torch.randint(-128, 127, (5, 5), dtype=torch.int8)
+        m = M(weight)
+        x = torch.randint(-128, 127, (5, 5), dtype=torch.int8)
+
+        gm = make_fx(m)(x)
+        self.assertTrue(torch.allclose(m(x), gm(x)))
+
+        gm = make_fx(torch.func.functionalize(M(weight)))(x)
+        self.assertTrue(torch.allclose(m(x), gm(x)))
+
+        FileCheck().check("torch.ops.higher_order.out_dtype").check("aten.mm.default").run(gm.code)
+        self.assertTrue(torch.allclose(m(x), gm(x)))
+        for node in gm.graph.nodes:
+            if node.op == "call_function" and node.target is out_dtype:
+                # Result of this node should be int32
+                self.assertTrue(node.meta["val"].dtype, torch.int32)
+                # Argument of this node should be int8
+                self.assertTrue(node.args[2].meta["val"].dtype, torch.int8)
+
+        x_casted = x.to(torch.int32)
+        weight_casted = weight.to(torch.int32)
+        numerical_res = torch.ops.aten.mm.default(x_casted, weight_casted)
```

Comment:
could you split the numerical test in a separate test? same for mul.Scalar, I feel it's better to separate the functional (make_fx, autograd etc.) tests and numerical tests